packages/client-adapters/tests/adapters/detectlang/adapter.test.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-env jest */
+/* eslint-disable no-unused-vars */
+import 'whatwg-fetch'
+
+import DetectLangAdapter from '@clAdapters/adapters/detectlang/adapter'
+
+describe('detectlang/adapter.test.js', () => {
+  console.error = function () {}
+  console.log = function () {}
+  console.warn = function () {}
+    
+  beforeEach(() => {
+    jest.spyOn(console, 'error')
+    jest.spyOn(console, 'log')
+    jest.spyOn(console, 'warn')
+  })
+  afterEach(() => {
+    jest.resetModules()
+  })
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  it('1 DetectLangAdapter - constructor uploads config, available', () => {
+    let adapter = new DetectLangAdapter({
+      category: 'detectlangGroup',
+      adapterName: 'detectlangMethod',
+      method: 'detectLanguage'
+    })
+  
+    expect(adapter.errors).toEqual([])
+    expect(adapter.l10n).toBeDefined()
+    expect(adapter.config).toBeDefined()
+  })
+
+  it.skip('2 DetectLangAdapter - getDetectedLangsList - real data', async () => {
+    let adapter = new DetectLangAdapter({
+      category: 'detectlangGroup',
+      adapterName: 'detectlangMethod',
+      method: 'detectLanguage'
+    })
+
+    const result = await adapter.getDetectedLangsList('Mare male')
+    expect(result).toEqual('eng')
+  }, 50000)
+
+  it('3 DetectLangAdapter - getDetectedLangsList - sourceData - full response', async () => {
+    let adapter = new DetectLangAdapter({
+      category: 'detectlangGroup',
+      adapterName: 'detectlangMethod',
+      method: 'detectLanguage',
+      sourceData: { data: {
+        detections: [
+          { language: 'en', isReliable: true, confidence: 3.36 },
+          { language: 'pt', isReliable: false, confidence: 3.36 },
+          { language: 'eu', isReliable: false, confidence: 3.36 }

```

Comment:
I'd recommend using a more complicated response where there are multiple `isReliable: false` and multiple `isReliable: true`, the highest confidence has `isReliable: false`, and the other ones have different `confidence`. That would test the language selection logic more fully without requiring  additional tests.

Reply:
thank you for the recomendation - done and method is optimized now :)